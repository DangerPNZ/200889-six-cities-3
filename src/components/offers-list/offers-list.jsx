import React from 'react';
import PropTypes from 'prop-types';
import {OfferCard} from '../offer-card/offer-card.jsx';

/* Временная функция. (магические значения не соответствуют критериям)
Использую во избежание антипаттерна с назначением в качестве id индексов итерируемого массива.
Применение на данном этапе пакета nanoid считаю избыточным.
Если необходимость в генерации id останется актуальной, подключу nanoid */
const getId = () => `_` + Math.random().toString(36).substr(2, 9);

export class OffersList extends React.PureComponent {
  constructor(props) {
    super(props);
    this.state = {
      activeItemData: null
    };
    this.onUpdateOffersListState = this.onUpdateOffersListState.bind(this);
  }
  render() {
    return (
      <div className="cities__places-list places__list tabs__content">
        {
          this.props.offersNames.map((name) => <OfferCard
            offerName = {name}
            onUpdateOffersListState = {this.onUpdateOffersListState}
            onUpdateAppState = {this.props.onUpdateAppState}
            key = {getId()}
          />)
        }
      </div>
    );
  }
  /* для обнуления состояния передать null */
  onUpdateOffersListState(activeItemData) {
    this.setState({
      activeItemData
    });
  }
}

OffersList.propTypes = {
  offersNames: PropTypes.arrayOf(PropTypes.string.isRequired).isRequired,
  onUpdateAppState: PropTypes.func.isRequired
};
