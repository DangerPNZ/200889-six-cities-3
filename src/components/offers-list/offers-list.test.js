import React from 'react';
import renderer from 'react-test-renderer';
import nanoid from 'nanoid';
import {OffersList} from './offers-list.jsx';

const REVIEWS_MOCK = [
  {
    author: `Hanna`,
    review: `Unique lightness of Amsterdam. The building is green and from 18th century.`,
    userRating: 1,
    date: `April 2017`
  },
  {
    author: `Bill`,
    review: `A quiet cozy and picturesque that.`,
    userRating: 0,
    date: `January 2018`
  },
  {
    author: `Ed`,
    review: `The building is green and from 18th century.`,
    userRating: 3,
    date: `September 2019`
  }
];
const TestDataValue = {
  OFFERS: [
    {
      name: `Canal View Prinsengracht`,
      coordinates: [52.3909553943508, 4.929309666406198],
      id: nanoid(),
      price: 100,
      type: `Hotel room`,
      premium: false,
      isFavorites: true,
      rating: 3.7,
      reviews: REVIEWS_MOCK
    },
    {
      name: `Nice, cozy, warm big bed apartment`,
      coordinates: [52.3809553943508, 4.939309666406198],
      id: nanoid(),
      price: 170,
      type: `Lux apartment`,
      premium: true,
      isFavorites: true,
      rating: 2.7,
      reviews: REVIEWS_MOCK
    }
  ]
};
const OFFERS_SORT_TYPE = `Price: low to high`;

it(`OffersList component structure test`, () => {
  const tree = renderer
  .create(
      <OffersList
        offers = {TestDataValue.OFFERS}
        onOfferHeadingClick = {() => {}}
        offersSortType = {OFFERS_SORT_TYPE}
        onOfferMouseInteract = {() => {}}
      />
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
