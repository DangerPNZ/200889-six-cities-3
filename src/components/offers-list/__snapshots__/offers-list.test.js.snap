// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OffersList component structure test 1`] = `
<div
  className="cities__places-list places__list tabs__content"
/>
`;
